client/console/src/main/java/org/apache/syncope/client/console/panels/search/ConnObjectSearchPanel.java

```diff
@@ -0,0 +1,122 @@
+import org.apache.syncope.common.lib.to.PlainSchemaTO;
+import org.apache.syncope.common.lib.to.ResourceTO;
+import org.apache.syncope.common.lib.types.AnyTypeKind;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.LoadableDetachableModel;
+
+public class ConnObjectSearchPanel extends AbstractSearchPanel {
+
+    private static final long serialVersionUID = 21020550706646L;
+
+    private final ConnectorRestClient connectorRestClient = new ConnectorRestClient();
+
+    protected final ResourceTO resource;
+
+    public static class Builder extends AbstractSearchPanel.Builder<ConnObjectSearchPanel> {
+
+        private static final long serialVersionUID = 6308997285778809578L;
+
+        protected final ResourceTO resource;
+
+        private final AnyTypeKind anyType;
+
+        private final String typeName;
+
+        public Builder(final ResourceTO resource, final AnyTypeKind anyType, final String type,
+                final IModel<List<SearchClause>> model) {
+            super(model);
+            this.resource = resource;
+            this.anyType = anyType;
+            this.typeName = type;
+        }
+
+        @Override
+        public ConnObjectSearchPanel build(final String id) {
+            return new ConnObjectSearchPanel(id, anyType, typeName, this);
+        }
+    }
+
+    protected ConnObjectSearchPanel(final String id, final AnyTypeKind kind, final Builder builder) {
+        super(id, kind, builder);
+        this.resource = builder.resource;
+    }
+
+    protected ConnObjectSearchPanel(final String id, final AnyTypeKind kind, final String type, final Builder builder) {
+        super(id, kind, type, builder);
+        this.resource = builder.resource;
+    }
+
+    @Override
+    protected void populate() {
+        this.types = new LoadableDetachableModel<List<SearchClause.Type>>() {
+
+            private static final long serialVersionUID = 22668815812716L;
+
+            @Override
+            protected List<SearchClause.Type> load() {
+                return getAvailableTypes();
+            }
+        };
+
+        this.dnames = new LoadableDetachableModel<List<String>>() {
+
+            private static final long serialVersionUID = 2989042618372L;
+
+            @Override
+            protected List<String> load() {
+                return getSchemas().map(PlainSchemaTO::getKey).collect(Collectors.toList());
+            }
+        };
+
+        this.anames = new LoadableDetachableModel<Map<String, PlainSchemaTO>>() {
+
+            private static final long serialVersionUID = 3002350300761L;
+
+            @Override
+            protected Map<String, PlainSchemaTO> load() {
+                return getSchemas().collect(Collectors.toMap(PlainSchemaTO::getKey, Function.identity(),
+                        (schema1, schema2) -> schema1));
+            }
+        };
+    }
+
+    protected List<SearchClause.Type> getAvailableTypes() {
```

Comment:
you can replace this method with `Collections.singletonList(SearchClause.Type.ATTRIBUTE)`